source/joker_effects.c

```diff
@@ -5,180 +5,206 @@
 #include "list.h"
 #include <stdlib.h>
 
-static JokerEffect default_joker_effect(Joker *joker, Card *scored_card) {
+
+#define SCORE_ON_CARD_SCORED_ONLY if (scored_when != JOKER_CALLBACK_ON_CARD_SCORED || !scored_card) { return effect; }
+#define SCORE_ON_CARD_HELD_ONLY   if (scored_when != JOKER_CALLBACK_ON_CARD_HELD   || !scored_card) { return effect; }
+#define SCORE_INDEPENDANT_ONLY    if (scored_when != JOKER_CALLBACK_INDEPENDANT)                    { return effect; }
```

Comment:
I don't like this. If the variable name changes from the function, this will result in confusing bugs from these macros. 

Ideally it would be designed to not use macros, but I see where theres a benefit here for reusability. 

I would suggest at the very least passing the names for `scored_when` and `effect`. There's 0 code change when it's built, but it makes more sense to the programmer, imo